workers/unity/Packages/com.improbable.gdk.tools/LocalLaunch.cs

```diff
@@ -0,0 +1,122 @@
+{
+    public static class LocalLaunch
+    {
+        private static readonly string
+            SpatialProjectRootDir = Path.GetFullPath(Path.Combine(Application.dataPath, "..", "..", ".."));
+
+        // Windows: The exit code is 0xc000013a when the user closes the console window, or presses Ctrl+C.
+        private const int WindowsCtrlCExitCode = -1073741510;
+
+        // Unix-like: The exit code is 128 + SIGINT (2).
+        private const int UnixSigIntExitCode = 128 + 2;
+
+        [MenuItem("Improbable/Spatial/Build worker configs")]
+        private static void BuildConfigMenu()
+        {
+            Debug.Log("Building worker configs...");
+            EditorApplication.delayCall += BuildConfig;
+        }
+
+        [MenuItem("Improbable/Spatial/Local launch")]
+        private static void LaunchMenu()
+        {
+            Debug.Log("Launching SpatialOS locally...");
+            EditorApplication.delayCall += () =>
+            {
+                BuildConfig();
+                Launch();
+            };
+        }
+
+        public static void BuildConfig()
+        {
+            using (new ShowProgressBarScope("Building worker configs..."))
+            {
+                // Run from the root of the project to build all available worker configs.
+                RedirectedProcess.RunIn(SpatialProjectRootDir, Common.SpatialBinary, "build", "build-config",
+                    "--json_output");
+            }
+        }
+
+        public static void Launch()
+        {
+            BuildConfig();
+            var processInfo = new ProcessStartInfo("spatial", "local launch")
+            {
+                CreateNoWindow = false,
+                UseShellExecute = true,
+                WorkingDirectory = SpatialProjectRootDir
+            };
+
+            var process = Process.Start(processInfo);
+
+            if (process == null)
+            {
+                return;
+            }
+
+            process.EnableRaisingEvents = true;
+            process.Exited += (sender, args) =>
+            {
+                // N.B. This callback is run on a different thread.
+                if (process.ExitCode == 0)
+                {
+                    return;
+                }
+
+                var logPath = Path.Combine(SpatialProjectRootDir, "logs");
+                var latestLogFile = Directory.GetFiles(logPath, "spatial_*.log")
+                    .Select(f => new FileInfo(f))
+                    .OrderBy(f => f.LastWriteTimeUtc).LastOrDefault();
+
+                if (latestLogFile == null)
+                {
+                    Debug.LogError($"Could not find a spatial log file in {logPath}.");
+                    return;
+                }
+
+                var content = File.ReadAllText(latestLogFile.FullName);
+                var message = $"Loaded from {latestLogFile.FullName}\n{content}";
+
+                if (WasProcessKilled(process))
+                {
+                    Debug.Log(message);
```

Comment:
Do we want to log the entire spatial log in the event you hit Ctrl + C?

Reply:
I wasn't sure; we could just print the path to the file instead of the whole thing

Reply:
Yeah I think that would be preferable - it will spam your console with things you probably don't care about.

Something like: `"spatial local launch" finished. View the log here - <log_path>`